executor/index_merge_reader.go

```diff
@@ -202,14 +206,15 @@ func (e *IndexMergeReaderExecutor) startPartialIndexWorker(ctx context.Context,
 	if err != nil {
 		return err
 	}
-
-	result, err := distsql.SelectWithRuntimeStats(ctx, e.ctx, kvReq, e.handleCols.GetFieldsTypes(), e.feedbacks[workID], getPhysicalPlanIDs(e.partialPlans[workID]), e.id)
+	result, err := distsql.SelectWithRuntimeStats(ctx, e.ctx, kvReq, e.handleCols.GetFieldsTypes(), e.feedbacks[workID], getPhysicalPlanIDs(e.partialPlans[workID]), e.getIndexID(workID))
 	if err != nil {
 		return err
 	}
 
 	result.Fetch(ctx)
 	worker := &partialIndexWorker{
+		idxMerge:     e,
```

Comment:
How a bout store `	stats      *IndexLookUpRunTimeStats` instead of `*IndexMergeReaderExecutor`?